element/test/index.js

```diff
@@ -212,4 +213,48 @@ describe( 'element', () => {
 			expect( element.prop( 'children' ) ).toBe( undefined );
 		} );
 	} );
+
+	describe( 'pure', () => {
+		it( 'functional component should rerender only when props change', () => {
+			let i = 0;
+			const MyComp = pure( () => {
+				return <p>{ ++i }</p>;
+			} );
+			const wrapper = mount( <MyComp /> );
+			wrapper.update(); // Upading don't rerender
```

Comment:
Typo in the comment (also in line 246).
How this `wrapper.update()` works, it's super confusing ...

Reply:
Well I wanted to rerender the component without passing new props or something. But it's not that important since passing a similar prop is also tested